x/stakeibc/keeper/msg_server_submit_tx.go

```diff
@@ -91,6 +94,79 @@ func (k Keeper) DelegateOnHost(ctx sdk.Context, hostZone types.HostZone, amt sdk
 	return nil
 }
 
+func (k Keeper) ReinvestRewards(ctx sdk.Context, hostZone types.HostZone) error {
+	_ = ctx
+	// the relevant ICA is the delegate account
+	owner := types.FormatICAAccountOwner(hostZone.ChainId, types.ICAAccountType_DELEGATION)
+	portID, err := icatypes.NewControllerPortID(owner)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "%s has no associated portId", owner)
+	}
+	connectionId, err := k.GetConnectionId(ctx, portID)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidChainID, "%s has no associated connection", portID)
+	}
+
+	// Fetch the relevant ICA
+	delegationAccount := hostZone.GetDelegationAccount()
+	withdrawAccount := hostZone.GetWithdrawalAccount()
+
+	SubmitTxs := k.SubmitTxs
+	GetParam := k.GetParam
+	cdc := k.cdc
+
+	var cb icqkeeper.Callback = func(k icqkeeper.Keeper, ctx sdk.Context, args []byte, query icqtypes.Query) error {
+		var msgs []sdk.Msg
+		queryRes := bankTypes.QueryAllBalancesResponse{}
+		err := cdc.Unmarshal(args, &queryRes)
+		if err != nil {
+			k.Logger(ctx).Error("Unable to unmarshal balances info for zone", "err", err)
+			return err
+		}
+		// Get denom dynamically
+		balance := queryRes.Balances.AmountOf(hostZone.HostDenom)
+		balanceDec := sdk.NewDec(balance.Int64())
+		commission := sdk.NewDec(int64(GetParam(ctx, types.KeyStrideCommission))).Quo(sdk.NewDec(100))
```

Comment:
Dividing by 100 is a bit of a hack here, we might be better off defining the commission as a Dec and passing it in directly. I just want to avoid any kind of rounding issues (I don't see obvious issues but maybe better to not manipulate/transform key parameters too much)

Reply:
Per @vish-stride comment in params.go
"you apparantly cannot safely encode floats, so we make commission * 100"
Given that, we need to divide by 100 somewhere. I think `Dec` is the best place to do so; the `Dec` type has 18 decimals so rounding errors shouldn't be an issue. More importantly, our commission is `10`, so 10/100 will produce a `Dec` type like `0.100000...`